src/interactivity/base/ServiceLocator.cpp

```diff
@@ -50,6 +50,14 @@ void ServiceLocator::RundownAndExit(const HRESULT hr)
     //   because doing so would prevent the render thread from progressing.
     if (locked.exchange(true, std::memory_order_relaxed))
     {
+        // Don't sleep forever with the console lock being held or otherwise
+        // no one can acquire it while we shut down.
+        const auto& gci = s_globals.getConsoleInformation();
+        while (gci.IsConsoleLocked())
+        {
+            gci.UnlockConsole();
+        }

```

Comment:
I put this in a safety measure so that `CloseConsoleProcessState()` works correctly even if the console lock was acquired. Do you think I should call the global `UnlockConsole()` instead so that this optionally calls `ProcessCtrlEvents`? I was hoping that this wouldn't be important given that this would be at least the second thread to enter `RundownAndExit`...

Reply:
Ah, I see. This is... i'm honestly not sure.
While we're running down and exiting _in response to a process exiting,_ I don't think we need to deliver it any more signals.